src/api/migraphx.py

```diff
@@ -220,6 +220,27 @@ def module(h):
              returns='migraphx::instruction_ref')
 
 
+@auto_handle()
+def value(h):
+    h.constructor('create')
+    h.method('is_null', returns='bool', const=True)
+    h.method('get_key', returns='const std::string', const=True)
+    cpp_types = ['int64_t', 'uint64_t', 'double', 'std::string', 'bool']
+    vt = ['int64', 'uint64', 'float', 'string', 'bool']
+    for vt, cpp_type in zip(vt, cpp_types):
+        if (vt == 'string'):
+            h.constructor('create_' + vt, api.params(i='const char*'))
+            h.constructor('create_' + vt + '_with_key',
+                          api.params(pkey='const char*', i='const char*'))
+        else:
+            h.constructor('create_' + vt, api.params(i=cpp_type))
+            h.constructor('create_' + vt + '_with_key',
+                          api.params(pkey='const char*', i=cpp_type))
+        h.method('if_' + vt, returns='const ' + cpp_type + '*', const=True)
```

Comment:
I dont think we need to expose the `if` functions through the C API. We can technically implement them through C++ by calling the `is` and `get` methods.

Reply:
`if` methods return pointer to original data in the struct. It is not possible to return pointer if i compose `get` and `is` method. 
1. I would be returning pointer to temp variable. 2. I can return by value, but in that case, we don't have default value (e.g. nullptr in case of return by pointer).

Reply:
Btw, I don't see `if` methods being used anywhere in the migraphx source.